torch/_inductor/fx_passes/freezing_patterns.py

```diff
@@ -148,6 +150,24 @@ def matmul_replacement(inp, w1, w2, w3):
         exclusive_arg_names=("w1", "w2", "w3"),
     )
 
+    def matmul_fuse_pattern_two(inp, w1, w2):
+        return (inp @ w1, inp @ w2)
+
+    def matmul_replacement_two(inp, w1, w2):
+        cat_t = torch.cat((w1, w2), dim=1)
+        mm = inp @ cat_t
+        return mm.chunk(2, dim=1)
```

Comment:
I think this assumes the sizes are equal.

Reply:
That condition is checked above:

https://github.com/pytorch/pytorch/blob/bfccf138b02bc4093d5990abdaac71c487842805/torch/_inductor/fx_passes/freezing_patterns.py#L130-L133

Reply:
Ah, it should be possible to handle the mismatched case too with a slice.

Reply:
will leave that for a follow up

Reply:
Sounds good, do add the test though.